codec-http/src/main/java/io/netty/handler/codec/http/HttpResponseStatus.java

```diff
@@ -453,15 +453,39 @@ public static HttpResponseStatus valueOf(int code) {
         return new HttpResponseStatus(code, reasonPhrase + " (" + code + ')');
     }
 
-    /**
-     * Parses the specified HTTP status line into a {@link HttpResponseStatus}.  The expected formats of the line are:
-     * <ul>
-     * <li>{@code statusCode} (e.g. 200)</li>
-     * <li>{@code statusCode} {@code reasonPhrase} (e.g. 404 Not Found)</li>
-     * </ul>
-     *
-     * @throws IllegalArgumentException if the specified status line is malformed
-     */
+    static HttpResponseStatus valueOf(int code, char[] chars, int startIndex, int endIndex) {
+        int size = endIndex - startIndex;
+        // We will try to servie 200 and 400 return codes via fast-path as these are the most common.
+        switch (code) {
+            // Fast-path for 200 OK
+            case 200:
+                if (size == 2 && chars[startIndex] == 'O' && chars[startIndex + 1] == 'K') {
```

Comment:
1. I'm a bit concerned about the reverse mapping being reliable. [rfc7231 section-6.1](https://tools.ietf.org/html/rfc7231#section-6.1) has the following:
```
The reason phrases listed here are only recommendations
   -- they can be replaced by local equivalents without affecting the
   protocol.
```

So I'm not sure what to do about this, but just wanted to bring it up.

2. Should we be using the `AsciiString` defines from [HttpResponseStatus](https://github.com/netty/netty/blob/master/codec-http/src/main/java/io/netty/handler/codec/http/HttpResponseStatus.java#L25) to keep these values consolidated?